volume.go

```diff
@@ -101,3 +103,68 @@ func (c *Client) RestoreSnapshot(dcid string, volid string, snapshotID string) (
 	err := c.DoWithRequest(req, resty.MethodPost, restoreSnapshotPath(dcid, volid), http.StatusAccepted)
 	return ret.GetHeader(), err
 }
+
+// CreateVolumeAndWait creates a volume and waits for the request to complete.
+// The default timeout is 5 minutes.
+func (c *Client) CreateVolumeAndWait(dcid string, request Volume, timeout time.Duration) (*Volume, error) {
+	volume, err := c.CreateVolume(dcid, request)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), DurationOrDefault(timeout, 5*time.Minute))
```

Comment:
So, imho we are going into a technical debt path here. For future version I already had a change in mind to change all function signatures to require context.Context. So I would say, that all new functions should expect a context so that the client does not have to take care of this.

Reply:
Yeah, that was a bit short-sighted. Using the caller's context is definitely better